### PR TITLE
Fetch individual threads

### LINKED ISSUE #587: chore(ui): PageData typing for chat page
The +page.svelte file at /chat/(dashboard)/[[thread_id]] receives page data with the ```export let data``` line towards the top of the file. The variable is correctly receiving the required data from the +layout.server.ts file two directories above (it needs the threads and messages).

The data in the layout is being fetched at a higher directory level so that several sub routes don't have to refetch it all the time and it would be desirable to keep it this way.

Normally, page data is automatically correctly typed (inferred) from the ```interface PageData``` in the app.d.ts file. SvelteKit will determine the type of the data based on the route location and that interface definition. For an unknown reason, it is not correctly typing the data with the +layout.server.ts data even though the data is indeed available.

 I opened a discussion on SvelteKit's Github here:
 https://github.com/sveltejs/kit/discussions/12269
 
 **This chore is to find a way to get that data into the pages with correct typing without over fetching.**

### PR DESCRIPTION
**Pull request by andrewrisse**
Refactor to fetch a thread when the thread is clicked on in the sidenav.

Previously, threads were retrieved from the threadsStore, but multiple pages triggering different "load" functions was causing state issues. This simplifies the process of keeping a thread up to date.

The threadsStore still loads all the threads when the app loads, this is useful for the sidenav (especially search). We do not re-fetch that data unless the entire app reloads, so it's a onetime large data fetch, then the app maintains the threads state internally as CRUD operations commence.

* Fixes some typing issues, as well as sets a better pattern for keeping threads up to date.
* Also fixes some edge cases where messages are not ordered correctly.
* Included some test cleanup

also closes #587 - page data typing

### CODE REVIEW

File: src/leapfrogai_ui/src/routes/chat/(dashboard)/[[thread_id]]/+page.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { browser } from '$app/environment';
+import type { LFThread } from '$lib/types/threads';
+import { threadsStore } from '$stores';
+
+export const load = async ({ params, fetch, depends }) => {
+  depends('lf:thread');
```
[the remark this answers is not in the document]
Comment (andrewrisse) [reply src/leapfrogai_ui/src/routes/chat/(dashboard)/[[thread_id]]/+page.ts:6]: depends gives you the ability to explicitly invalidate a load function.
For example, if I wanted to re-run the load function for a page that was "tagged" with depends("lf:thread"), I could call invalidate("lf:thread"), and it would re-run.
In this case, we aren't invalidating the thread load function anywhere. I think at one point I was, but then removed the invalidate call. I will remove this depends.

We use this pattern in other places like invalidate('lf:assistants')

File: src/leapfrogai_ui/src/routes/chat/(dashboard)/[[thread_id]]/+page.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  const promises = [fetch('/api/assistants'), fetch('/api/files')];
+
+  if (params.thread_id) promises.push(fetch(`/api/threads/${params.thread_id}`));
+
+  const promiseResponses = await Promise.all(promises);
+
+  const assistants = await promiseResponses[0].json();
+  const files = await promiseResponses[1].json();
```
[the remark this answers is not in the document]
Comment (andrewrisse) [reply src/leapfrogai_ui/src/routes/chat/(dashboard)/[[thread_id]]/+page.ts:14]: yes, the await Promise.all() makes all the async requests to the api which returns Response objects. The Response objects have the .json() function that is also async.

File: src/leapfrogai_ui/src/routes/chat/(dashboard)/[[thread_id]]/+page.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { browser } from '$app/environment';
+import type { LFThread } from '$lib/types/threads';
+import { threadsStore } from '$stores';
+
+export const load = async ({ params, fetch, depends }) => {
+  depends('lf:thread');
+  const promises = [fetch('/api/assistants'), fetch('/api/files')];
+
+  if (params.thread_id) promises.push(fetch(`/api/threads/${params.thread_id}`));
```
[the remark this answers is not in the document]
Comment (andrewrisse) [reply src/leapfrogai_ui/src/routes/chat/(dashboard)/[[thread_id]]/+page.ts:9]: yes, if you visit /chat without a thread_id in the url, you still hit this page and this load function is called.

File: src/leapfrogai_ui/src/routes/chat/(dashboard)/[[thread_id]]/+page.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import { browser } from '$app/environment';
+import type { LFThread } from '$lib/types/threads';
+import { threadsStore } from '$stores';
+
+export const load = async ({ params, fetch, depends }) => {
+  depends('lf:thread');
+  const promises = [fetch('/api/assistants'), fetch('/api/files')];
```
[the remark this answers is not in the document]
Comment (andrewrisse) [reply src/leapfrogai_ui/src/routes/chat/(dashboard)/[[thread_id]]/+page.ts:7]: Files don't have any association with threads right now, only assistants.
If you had a selected assistant, we could fetch that assistant's vector store, get the file ids, then fetch only those files, but it would have to be a one by one fetch for each of the files. I think this would be cumbersome and only marginally beneficial because the list files call only returns metadata about the file, not the actual file, so we aren't getting too much data here.

**Review by YrrepNoj — APPROVED** [review]
LGTM
